src/hooks/useBackButton.tsx

```diff
@@ -29,7 +37,7 @@ export default function useBackButton({ username }: Props) {
         void replace(`/${username}`);
```

Comment:
this should be controlled by the arguments coming into this hook right? i.e. if you're on the NFT Detail Page, and you came from the Collection Page, the back button should take you to your collection page (not user profile)

Reply:
I believe this is ok atm because this else block is a fallback only if canGoBack is falsy or back is disabled (came from edit route). When clicking this back button from the NFT Detail Page after they arrived from the Collection Page, `canGoBack` is true so `back()` will take the user to the correct page.

If they arrive at the NFT Detail Page directly so they have no history, then they this will be called and they will be taken to the Gallery Page.